src/slice.rs

```diff
@@ -1229,6 +1370,53 @@ impl<'a> RopeSlice<'a> {
         }
     }
 
+    /// Non-panicking version of [`slice()`](RopeSlice::slice).
+    pub fn get_byte_slice<R>(&self, byte_range: R) -> Option<RopeSlice<'a>>
+    where
+        R: RangeBounds<usize>,
+    {
+        let (start, end) = {
+            let start_range = start_bound_to_num(byte_range.start_bound());
+            let end_range = end_bound_to_num(byte_range.end_bound());
+
+            // Early-out shortcut for taking a slice of the full thing.
+            if start_range == None && end_range == None {
+                return Some(*self);
+            }
+
+            (
+                start_range.unwrap_or(0),
+                end_range.unwrap_or_else(|| self.len_chars()),
+            )
+        };
+
+        // Bounds check
+        if start <= end && end <= self.len_chars() {
+            match *self {
+                RopeSlice(RSEnum::Full {
+                    node, start_info, ..
+                }) => Some(RopeSlice::new_with_byte_range(
+                    node,
+                    start_info.bytes as usize + start,
+                    start_info.bytes as usize + end,
+                )),
+                RopeSlice(RSEnum::Light { text, .. }) => {
+                    let start_byte = start;
+                    let end_byte = end;
```

Comment:
I kept the name here but this could be cleaned up by renaming the variables further up